Data/Root/RootDataIO.cpp

```diff
@@ -26,39 +26,38 @@ RootDataIO::RootDataIO(const std::string TreeName_, int NumberEventsToProcess_)
 
 std::vector<ComPWA::Event>
 RootDataIO::readData(const std::string &InputFilePath) const {
-  TFile File(InputFilePath.c_str());
-  if (File.IsZombie())
+  auto chain = new TChain(TreeName.c_str());
+  chain->Add(InputFilePath.c_str());
+  if (!chain->GetListOfFiles()->GetEntriesFast())
     throw std::runtime_error("RootDataIO::RootDataIO() | "
-                             "Can't open data file: " +
+                             "Unable to load files: " +
                              InputFilePath);
 
-  TTree *fTree = (TTree *)File.Get(TreeName.c_str());
-
-  if (!fTree)
+  if (!chain->GetEntriesFast())
```

Comment:
You don't check the chain for nullptr here. So this line will segfault, if the tree can't be read. So you should use `if(!chain)`. Maybe it would also be better not to throw an error, but just log an error. And return an empty vector.

Reply:
No, `TChain` works differently than `TTree` in this regard: you don't `Get` a `TChain` from a `TFile`, but (much like `RootDataIO`) define before hand what the name of the `TTree` is when you construct the `TChain`, then `Add` files.